osu.Game/Beatmaps/IBeatmap.cs

```diff
@@ -24,7 +24,7 @@ public interface IBeatmap : IJsonSerializable
         /// <summary>
         /// The control points in this beatmap.
         /// </summary>
-        ControlPointInfo ControlPointInfo { get; }
+        ControlPointInfo ControlPointInfo { get; set; }
```

Comment:
I can look at this, but I'm not sure this will work out (there's a lot of very local logic being done at the moment for the pieces are are "ready" to make copies of... some code relies on the references being the same).

Reply:
Hmm, maybe I misspoke in the comment - I think I'd be fine with only deep cloning the control points inside `Clone()`, the rest can be as shallow as it currently is. That said sounds like we're in a bit of trouble if we're relying on reference equality in places.

Reply:
I guess my main concern is stuff like `BeatmapInfo` which isn't cloned (except for specific cases see https://github.com/ppy/osu/blob/97879c3c98bcc19012c52e86287b2b8c8f0357ab/osu.Game/Beatmaps/WorkingBeatmap.cs#L126-L127)

But yeah, maybe it can be the first step in the right direction by moving into the clone impl.

Reply:
I've made the change, see if it's what you expect.